pkg/plugins/plugins.go

```diff
@@ -0,0 +1,86 @@
+package plugins
+
+import (
+	"errors"
+	"sync"
+
+	"github.com/Sirupsen/logrus"
+)
+
+var (
+	ErrNotImplements = errors.New("Plugin not implements particular driver")
+)
+
+type plugins struct {
+	sync.Mutex
+	plugins map[string]*Plugin
+}
+
+var storage = plugins{plugins: make(map[string]*Plugin)}
+
```

Comment:
On the other hand, if drivers are loaded when they are mentioned, it makes dependencies easier to account for. It is more easy to reconcile the race between the plugin loading and registering itself as a driver, and starting a container that mentions the driver.

If a container is started that mentions an unknown driver, what should libnetwork do? It could fail, making the start-up ordering something of a crap-shoot; or it could wait for a plugin that may never come along.